mm2src/rpc_task/src/manager.rs

```diff
@@ -103,10 +107,26 @@ impl<Task: RpcTask> RpcTaskManager<Task> {
         Some(rpc_status)
     }
 
-    pub fn new_shared() -> RpcTaskManagerShared<Task> { Arc::new(Mutex::new(Self::default())) }
+    pub fn new(streaming_manager: StreamingManager) -> Self {
+        RpcTaskManager {
+            tasks: HashMap::new(),
+            streaming_manager,
+        }
+    }
+
+    pub fn new_shared(streaming_manager: StreamingManager) -> RpcTaskManagerShared<Task> {
+        Arc::new(Mutex::new(Self::new(streaming_manager)))
+    }
 
     pub fn contains(&self, task_id: TaskId) -> bool { self.tasks.contains_key(&task_id) }
 
+    fn get_client_id(&self, task_id: TaskId) -> Option<u64> {
```

Comment:
optionalaity here is for that this call might fail (not find a client id), not that the client id is optional.
if the task status is ok, error, canceling (cancelling is some transitional state needed for trezor code), these states don't store a client id (for simplicity).

ok doesn't need the client id since this state means the task has finished and we have already sent the result to the client.
error is the same.